src/Components/Modals/DisableRule.spec.ct.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { mount } from '@cypress/react';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+
+import DisableRule from './DisableRule';
+import { Intl } from '../../Utilities/intlHelper';
+import getStore from '../../Store';
+import '@patternfly/patternfly/patternfly.scss';
+import {
+  ouiaId,
+  CHECKBOX,
+  TEXT_INPUT,
+} from '../../../cypress/utils/components';
+
+const MODAL = ouiaId('recommendation-disable');
+const SAVE_BUTTON = ouiaId('confirm');
+const CANCEL_BUTTON = ouiaId('cancel');
+
+describe('modal without hosts', () => {
+  beforeEach(() => {
+    mount(
+      <MemoryRouter>
+        <Intl>
+          <Provider store={getStore()}>
+            <DisableRule
+              isModalOpen={true}
+              rule={{ rule_id: 'abc', disabled: false }}
+            />
+          </Provider>
+        </Intl>
+      </MemoryRouter>
+    );
+
+    cy.intercept('POST', '/api/insights-results-aggregator/v2/ack', {
+      statusCode: 200,
+    }).as('ackRequest');
+  });
+
+  it('exits', () => {
+    cy.get(MODAL).should('exist');
+    cy.get(CHECKBOX).should('not.exist');
+
+    // TODO check that no request is send
+
+    // TODO check page is reloaded afterwards
```

Comment:
but does not it trigger any kind of network request for that? Or a JS call that we can spy?

Reply:
Which scenario are you gonna test here? I don't get the goal of this `it`. Is it supposed to check if the modal exists and renders?

Reply:
I think I placed the comment in the wrong `it`. The idea is that after a the modal triggers an action in the backend, it should also trigger an action in frontend (e.g. refresh table). I would like to check that such action/callback gets triggered. Can be done easily?

Reply:
@ikerreyes  in this case, you can 1) catch API requests: when, using the modal, you are disabling a rule only for one cluster, then wait for `v1/.../disable` request. If disable globally => `v2/ack` request; 2) pass `afterFn` parameter to the modal component as a cypress stub. It should be called once network request succeed.

Testing whether a table is updated is somehow impossible in your specs here: you mount only the modal, not a table

Reply:
Yes, we need to check whether the disablement request doesn't result in error,  because otherwise this function won't be called. If it results in error (404?), then let's intercept it. After all these checks, from the https://github.com/RedHatInsights/insights-rbac-ui/pull/1099 code, the `afterFn` will be called 👍🏼